version: '3'

services:
  postgres:
    image: postgres:13.1
    healthcheck:
      test: [ "CMD", "pg_isready", "-q", "-d", "postgres", "-U", "root" ]
      timeout: 45s
      interval: 10s
      retries: 10
    restart: always
    environment:
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - APP_DB_USER=${APP_DB_USER}
      - APP_DB_PASS=${APP_DB_PASS}
      - APP_DB_NAME=${APP_DB_NAME}
    volumes:
      - ./db:/docker-entrypoint-initdb.d
    ports:
      - 5432:5432
    extra_hosts:
      - "host.docker.internal:172.17.0.1"
  datascience-notebook:
      image: jupyter/datascience-notebook
      environment:
        - POSTGRES_USER=${POSTGRES_USER}
        - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
        - APP_DB_USER=${APP_DB_USER}
        - APP_DB_PASS=${APP_DB_PASS}
        - APP_DB_NAME=${APP_DB_NAME}
      volumes:
        - ./notebook:/home/jovyan/work
      ports:
        - 8888:8888
      container_name:  datascience-notebook-container